source/plugins/ruby/ApplicationInsightsUtility.rb

```diff
@@ -21,10 +21,15 @@ class ApplicationInsightsUtility
   @@EnvApplicationInsightsEndpoint = "APPLICATIONINSIGHTS_ENDPOINT"
   @@EnvControllerType = "CONTROLLER_TYPE"
   @@EnvContainerRuntime = "CONTAINER_RUNTIME"
-
+  @@isWindows = false
+  @@hostName = (OMS::Common.get_hostname)
+  @@os_type = ENV["OS_TYPE"]
+  if !@@os_type.nil? && !@@os_type.empty? && @@os_type.strip.casecmp("windows") == 0
+    @@isWindows = true
+    @@hostName = ENV["HOSTNAME"]
```

Comment:
Looks like this doesnt give windows host name, right? since AI telemetry code running on rs node?

Reply:
This is set in the yaml - https://github.com/microsoft/Docker-Provider/blob/959b455d5ab873b6fa5ed7445bd61dc847ec2c08/kubernetes/omsagent.yaml#L779